langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal

```diff
@@ -594,5 +594,49 @@ function assert(anydata actual, anydata expected) {
     }
 }
 
+function testData() {
+    xml:Element authors = xml `<authors><author><name>Enid<middleName/>Blyton</name></author></authors>`;
+    assertEquals(authors.data(), "EnidBlyton");
+
+    xml:Text text = xml `hello&gt;abc`;
+    assertEquals(text.data(), "hello>abc");
+
+    xml:Comment cmnt = xml `<!-- this is a comment -->`;
+    assertEquals(cmnt.data(), "");
 
+    xml:ProcessingInstruction pi = xml `<?ins key=val other=val?>`;
+    assertEquals(pi.data(), "");
+
+    xml x = xml `<elem>&lt;</elem>`;
+    assertEquals((x/*).data(), "<");
+
+    xml concat = authors + text + cmnt + pi + x;
+    assertEquals(concat.data(), "EnidBlytonhello>abc<");
+}
 
+function testXmlGetContentOverASequence() {
```

Comment:
Shall we add similar tests for PIs too?

Reply:
Added https://github.com/ballerina-platform/ballerina-lang/pull/29047/commits/e08a8da23febf77ed9aab1be2a6a4e4e6ca567c4